packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -26,226 +24,93 @@ const getClassNames = classNamesFunction<ISwatchColorPickerStyleProps, ISwatchCo
 
 const COMPONENT_NAME = 'SwatchColorPicker';
 
-export class SwatchColorPickerBase extends React.Component<ISwatchColorPickerProps, ISwatchColorPickerState> {
-  public static defaultProps = {
-    cellShape: 'circle',
-    disabled: false,
-    shouldFocusCircularNavigate: true,
-    cellMargin: 10,
-  } as ISwatchColorPickerProps;
-
-  private _id: string;
-  private _cellFocused: boolean;
-
-  private navigationIdleTimeoutId: number | undefined;
-  private isNavigationIdle: boolean;
-  private readonly navigationIdleDelay: number = 250 /* ms */;
-  private async: Async;
+export const SwatchColorPickerBase = (props: ISwatchColorPickerProps) => {
+  const [selectedIndex, setSelectedIndex] = React.useState();
+  const id = props.id || getId('swatchColorPicker');
+  let cellFocused: boolean;
+  let currentSelectedIndex: number | undefined = undefined;
+
+  const {
+    colorCells,
+    cellShape = 'circle',
+    columnCount,
+    // tslint:disable:deprecation
+    ariaPosInSet = props.positionInSet,
+    ariaSetSize = props.setSize,
+    // tslint:enable:deprecation
+    shouldFocusCircularNavigate = true,
+    className,
+    disabled = false,
+    doNotContainWithinFocusZone,
+    styles,
+    cellMargin = 10,
+  } = props;
+
+  const classNames = getClassNames(styles!, {
+    theme: props.theme!,
+    className,
+    cellMargin,
+  });
 
   // Add an index to each color cells. Memoizes this so that color cells do not re-render on every update.
-  private _getItemsWithIndex = memoizeFunction((items: IColorCellProps[]) => {
+  const getItemsWithIndex = memoizeFunction((items: IColorCellProps[]) => {
```

Comment:
This can go outside the component body (and convert the comment to a doc comment)